### PR TITLE
allow determining/setting register pointed to by the unnamed ("")

### PR DESCRIPTION
**Pull request by andymass**
As you know, the unnamed `""` register behaves as a pointer to the most recently used register (this is most often the yank register `"0`).  However, there is no way to find out which register `""` is currently pointing to.  There is also no way to change it (`setreg("", ...)` always sets it to `"0`).  Historically scripts often save/restore registers which is improved with `getregtype`, but that still destroys whatever pointer is set.

This PR adds `getreginfo()` and `setreg(..., Dictionary)`.

~This PR adds `getregpoint()` and `setreg(..., ..., 'p')` flag.  An example for `setreg`: `call setreg('a', 'contents', 'p')` will set register `"a` to string 'contents' and additionally set `""` to point to `"a`.  Naming of functions is up for discussion.  Unfortunately, there's no way to change the pointer without also specifying contents (one can use `getreg`), but this flag avoids adding a second additional function.~

### CODE REVIEW

**Comment by ZyX-I** [comment]
This looks like something which makes saving/restoring register even more complicated. I would rather suggest to add `getreginfo(regname)` function which will return a dictionary, containing everything needed to restore register, specifically:

- Register contents. It is essential that it *must* be a list of strings, you can’t restore registers which have embedded NUL otherwise.
- Register type: what `getregtype()` returns.
- Boolean flag which determines whether given register is (not) aliased to `"`.

Then to restore a single register (obviously unless &clipboard contains unnamedsomething: this is impossible to restore properly) one would need to call `setreg(regname, dict)`: no additions to `options` flags are needed as `getreginfo` returns a dictionary (i.e. not one of the types which were already accepted) and register type is saved inside a dictionary. Also should register gain some other settable properties (e.g. not sure about Vim, but Neovim *does* have timestamp for ShaDa and Vim *did* update viminfo with similar functionality, though timestamp is currently not restorable) they can simply be added to that dictionary.

**Comment by ZyX-I** [comment]
(Also for the `"` register itself dictionary should have different key with a name of register `"` is aliased to attached.)

**Comment by andymass** [comment]
Updated for latest vim.  Context:
> Patch to add getreginfo() and setreg() with an option to set the unnamed register "", So that registers can be saved and fully restored.

**Comment by lacygoill** [comment]
@andymass When invoking `setreg()` with 3 arguments, we can set the type of a register to characterwise using either `v` or `c`:

    vim -es -Nu NONE -i NONE -S <(cat <<'EOF'
      call setreg('"', ['foo', 'bar'], 'c')
      "                                 ^
      "                                 ✔
      pu=getreginfo('\"')->string()
      %p
      qa!
    EOF
    )

    {'regcontents': ['foo', 'bar'], 'regtype': 'v', 'points_to': '0'}
                                                ^

But if we invoke `setreg()` with 2 arguments, we can no longer set the type of a register to characterwise using `c`.  We have to use `v`, otherwise the type is automatically set to `V`:

    vim -es -Nu NONE -i NONE -S <(cat <<'EOF'
      let info = {'regcontents': ['foo', 'bar'], 'regtype': 'c'}
      "                                                      ^
      "                                                      ✘
      call setreg('"', info)
      pu=getreginfo('\"')->string()
      %p
      qa!
    EOF
    )

    {'regcontents': ['foo', 'bar'], 'regtype': 'V', 'points_to': '0'}
                                                ^

Is it working as intended, and we should never use `c`, `l`, `b` to specify the type of a register when using a dictionary as the 2nd argument to `setreg()`?

**Comment by andymass** [comment]
@lacygoill, I could see either way.  Maybe we shouldn't support the old style?  At the very least it should raise an error I think.

**Comment by lacygoill** [comment]
> Maybe we shouldn't support the old style?

I find `c`, `l`, `b` more readable (easier to distinguish `c` from `l` than `v` from `V`), but I don't have a strong opinion on the matter, so whatever you choose is fine by me.

> At the very least it should raise an error I think.

Strongly agree.

**Comment by andymass** [comment]
Hmm, I noticed another (quite) minor bug.  Apparently, `@"` kind of internally points to `0` on startup, but not in a normal way..

    ./vim --clean +"let @0=0 | echo getreginfo('0') | echo getreginfo('\"')"

    {'regcontents': ['0'], 'isunnamed': v:false, 'regtype': 'v'}
    {'regcontents': ['0'], 'regtype': 'v', 'points_to': '"'}

**Comment by andymass** [comment]
A patch for the original problem @lacygoill pointed out and patch which makes `setreg('a', {})` should clear the register.

```diff
diff --git a/src/evalfunc.c b/src/evalfunc.c
index 667888375..0b2505889 100644
--- a/src/evalfunc.c
+++ b/src/evalfunc.c
@@ -7268,6 +7268,37 @@ f_setpos(typval_T *argvars, typval_T *rettv)
     }
 }
 
+/*
+ * Translate a register type string to the yank type and block length
+ */
+    static int
+get_yank_type(char_u **pp, char_u *yank_type, long *block_len)
+{
+    char_u *stropt = *pp;
+    switch (*stropt)
+    {
+	case 'v': case 'c':	// character-wise selection
+	    *yank_type = MCHAR;
+	    break;
+	case 'V': case 'l':	// line-wise selection
+	    *yank_type = MLINE;
+	    break;
+	case 'b': case Ctrl_V:	// block-wise selection
+	    *yank_type = MBLOCK;
+	    if (VIM_ISDIGIT(stropt[1]))
+	    {
+		++stropt;
+		*block_len = getdigits(&stropt) - 1;
+		--stropt;
+	    }
+	    break;
+	default:
+	    return FAIL;
+    }
+    *pp = stropt;
+    return OK;
+}
+
 /*
  * "setreg()" function
  */
@@ -7302,30 +7333,30 @@ f_setreg(typval_T *argvars, typval_T *rettv)
     if (argvars[1].v_type == VAR_DICT)
     {
 	dict_T	    *d = argvars[1].vval.v_dict;
-	dictitem_T  *di = dict_find(d, (char_u *)"regcontents", -1);
+	dictitem_T  *di;
+
+	if (d == NULL || d->dv_hashtab.ht_used == 0)
+	{
+	    // Empty dict, clear the register (like setreg(0, []))
+	    char_u *lstval[2] = {NULL, NULL};
+	    write_reg_contents_lst(regname, lstval, 0, FALSE, MAUTO, -1);
+	    return;
+	}
+
+	di = dict_find(d, (char_u *)"regcontents", -1);
 	if (di != NULL)
 	    regcontents = &di->di_tv;
 
 	stropt = dict_get_string(d, (char_u *)"regtype", FALSE);
 	if (stropt != NULL)
-	    switch (*stropt)
+	{
+	    int ret = get_yank_type(&stropt, &yank_type, &block_len);
+	    if (ret == FAIL || *++stropt != NUL)
 	    {
-		case 'v':		// character-wise selection
-		    yank_type = MCHAR;
-		    break;
-		case 'V':		// line-wise selection
-		    yank_type = MLINE;
-		    break;
-		case Ctrl_V:		// block-wise selection
-		    yank_type = MBLOCK;
-		    if (VIM_ISDIGIT(stropt[1]))
-		    {
-			++stropt;
-			block_len = getdigits(&stropt) - 1;
-			--stropt;
-		    }
-		    break;
+		semsg(_(e_invargval), "value");
+		return;
 	    }
+	}
 
 	if (regname == '"')
 	{
@@ -7344,6 +7375,12 @@ f_setreg(typval_T *argvars, typval_T *rettv)
 
     if (argvars[2].v_type != VAR_UNKNOWN)
     {
+	if (yank_type != MAUTO)
+	{
+	    semsg(_(e_toomanyarg), "setreg");
+	    return;
+	}
+
 	stropt = tv_get_string_chk(&argvars[2]);
 	if (stropt == NULL)
 	    return;		// type error
@@ -7353,21 +7390,8 @@ f_setreg(typval_T *argvars, typval_T *rettv)
 		case 'a': case 'A':	// append
 		    append = TRUE;
 		    break;
-		case 'v': case 'c':	// character-wise selection
-		    yank_type = MCHAR;
-		    break;
-		case 'V': case 'l':	// line-wise selection
-		    yank_type = MLINE;
-		    break;
-		case 'b': case Ctrl_V:	// block-wise selection
-		    yank_type = MBLOCK;
-		    if (VIM_ISDIGIT(stropt[1]))
-		    {
-			++stropt;
-			block_len = getdigits(&stropt) - 1;
-			--stropt;
-		    }
-		    break;
+		default:
+		    get_yank_type(&stropt, &yank_type, &block_len);
 	    }
     }
 
diff --git a/src/testdir/test_registers.vim b/src/testdir/test_registers.vim
index 627799c49..c9135628c 100644
--- a/src/testdir/test_registers.vim
+++ b/src/testdir/test_registers.vim
@@ -485,6 +485,14 @@ func Test_set_register_dict()
   call assert_equal(['six'], getreginfo('0').regcontents)
   call assert_equal(['six'], getreginfo('"').regcontents)
 
+  let @x = 'one'
+  call setreg('x', {})
+  call assert_equal(1, len(split(execute('reg x'), '\n')))
+
+  call assert_fails("call setreg('0', #{regtype: 'V'}, 'v')", 'E118:')
+  call assert_fails("call setreg('0', #{regtype: 'X'})", 'E475:')
+  call assert_fails("call setreg('0', #{regtype: 'vy'})", 'E475:')
+
   bwipe!
 endfunc
```

**Comment by lacygoill** [comment]
Can confirm that the patch fixes the first issue; i.e. one can now specify the type of a register using `c`, `l`, `b`.  Which makes it more consistent with how you can specify a register type via `setreg()`'s third argument, and with how it's referred to in the output of `:registers` or `:display`.

---

OTOH, the second issue is still there:

    ./src/vim -Nu NONE -i NONE +"let @0=0 | echo getreginfo('0') | echo getreginfo('\"')"

                                        vvvvvvv
    {'regcontents': ['0'], 'isunnamed': v:false, 'regtype': 'v'}
    {'regcontents': ['0'], 'regtype': 'v', 'points_to': '"'}
                                                         ^


I would expect this output instead:

                                        vvvvvv
    {'regcontents': ['0'], 'isunnamed': v:true, 'regtype': 'v'}
    {'regcontents': ['0'], 'regtype': 'v', 'points_to': '0'}
                                                         ^

But maybe it's because I don't understand well enough how registers work on startup.

**Comment by lacygoill** [comment]
> OTOH, the second issue is still there:

>     ./src/vim -Nu NONE -i NONE +"let @0=0 | echo getreginfo('0') | echo getreginfo('\"')"

>                                         vvvvvvv
>     {'regcontents': ['0'], 'isunnamed': v:false, 'regtype': 'v'}
>     {'regcontents': ['0'], 'regtype': 'v', 'points_to': '"'}
>                                                          ^


> I would expect this output instead:

>     {'regcontents': ['0'], 'isunnamed': v:false, 'regtype': 'v'}
>     {'regcontents': ['0'], 'regtype': 'v', 'points_to': '0'}
>                                                          ^

> But maybe it's because I don't understand well enough how registers work on startup.

Never mind, what I said was completely wrong.

**Comment by lacygoill** [comment]
@andymass: Sorry to bother you again, but I've noticed something else which could maybe be improved.

I think it would be useful and more consistent for `setreg()` to accept a dictionary containing only a `regtype` key:

    call setreg('r', #{regtype: 'l'})

Right now, such a dictionary is ignored:

    $ vim -Nu NONE -i NONE
    :call setreg('r', 'r register', 'c')
    :call setreg('r', #{regtype: 'l'})
    :reg r

    Type Name Content
      c  "r   r register
      ^
      ✘

Notice that the type of the register was not reset from characterwise to linewise.

---

It would be useful to reset the type of a register.  The alternatives are either more verbose/less intuitive or unreliable.

This is more verbose/less intuitive:

    call getreginfo('r')->extend({'regtype': 'V'})->setreg('r')

And this is unreliable:

    call setreg('r', [''], 'aV')

First, if you try to alter the type of the unnamed register like this, it will be automatically reconnected to `"0`.

    $ vim -Nu NONE -i NONE +"pu!='text'" +'norm! "ryy'
    :echo getreginfo('"').points_to
    r

    :call setreg('"', [''], 'av')
    :echo getreginfo('"').points_to
    0

Second, if you – accidentally – reset a linewise register into a linewise register, an undesirable extra newline is appended:

    $ vim -Nu NONE -i NONE +"pu!='text'" +'norm! yy'
    :call setreg('"', [''], 'aV')
    :reg "
    l  ""   text^J^J
                  ^^
                  ✘

See https://github.com/vim/vim/issues/323.

---

And it would make the usage of `setreg()` more consistent.

Right now, `setreg()` already accepts a dictionary containing only a `regcontents` key:

    call setreg('r', #{regcontents: 'new contents'})

Example:

    $ vim -Nu NONE -i NONE
    :call setreg('r', 'r register', 'c')
    :call setreg('r', #{regcontents: 'new contents'})
    :reg r

    Type Name Content
      c  "r   new contents
              ^----------^

And it also accepts a dictionary containing only an `isunnamed` key:

    call setreg('r', #{isunnamed: v:true})

Example:

    $ vim -Nu NONE -i NONE
    :call setreg('"', 'unnamed')
    :pu
    unnamed

    :undo
    :call setreg('r', 'r register', 'c')
    :call setreg('r', #{isunnamed: v:true})
    :pu
    r register

But it ignores a dictionary containing only a `regtype` key.

---

What do you think?

**Comment by andymass** [comment]
@lacygoill I definitely agree, although the register code on whole is a big mess and it's difficult to do anything in it.  Anyway, I have had a function and maps to change the register's type in my vimrc for a very long time, so I see the usefulness.

Here is my best guess of a patch.  Seems to work, but if you have the ability to test it, especially relative the problems you pointed out, that would be really great.

```diff
diff --git a/src/evalfunc.c b/src/evalfunc.c
index 53584dc46..3ba76db56 100644
--- a/src/evalfunc.c
+++ b/src/evalfunc.c
@@ -7456,6 +7456,8 @@ free_lstval:
 	write_reg_contents_ex(regname, strval, -1,
 						append, yank_type, block_len);
     }
+    else if (yank_type != MAUTO)
+	set_reg_type(regname, yank_type, block_len);
     if (pointreg != 0)
 	get_yank_register(pointreg, TRUE);
 
diff --git a/src/proto/register.pro b/src/proto/register.pro
index 07a74b352..698e5dac7 100644
--- a/src/proto/register.pro
+++ b/src/proto/register.pro
@@ -5,6 +5,7 @@ yankreg_T *get_y_current(void);
 yankreg_T *get_y_previous(void);
 void set_y_current(yankreg_T *yreg);
 void set_y_previous(yankreg_T *yreg);
+void set_reg_type(int regname, char_u yank_type, long block_len);
 int get_expr_register(void);
 void set_expr_line(char_u *new_line);
 char_u *get_expr_line(void);
diff --git a/src/register.c b/src/register.c
index 66dd0cca6..110847c2f 100644
--- a/src/register.c
+++ b/src/register.c
@@ -75,6 +75,21 @@ set_y_previous(yankreg_T *yreg)
 }
 
 #if defined(FEAT_EVAL) || defined(PROTO)
+/*
+ * Set the register's type (and block length, for MBLOCK)
+ * without modifying the register contents.
+ */
+    void
+set_reg_type(int regname, char_u yank_type, long block_len)
+{
+    get_yank_register(regname, FALSE);
+
+    y_current->y_type = yank_type;
+    if (yank_type == MBLOCK)
+	y_current->y_width = block_len;
+    y_current->y_time_set = vim_time();
+}
+
 /*
  * Keep the last expression line here, for repeating.
  */
diff --git a/src/testdir/test_registers.vim b/src/testdir/test_registers.vim
index d92ad977f..d41cc4510 100644
--- a/src/testdir/test_registers.vim
+++ b/src/testdir/test_registers.vim
@@ -492,6 +492,18 @@ func Test_set_register_dict()
   call assert_fails("call setreg('0', #{regtype: 'V'}, 'v')", 'E118:')
   call assert_fails("call setreg('0', #{regtype: 'X'})", 'E475:')
   call assert_fails("call setreg('0', #{regtype: 'vy'})", 'E475:')
+  call assert_fails("call setreg('0', #{regtype: 'a'})", 'E475:')
+
+  call setreg('x', ['one'], 'b52')
+  call setreg('x', #{regtype: 'b52'})
+  call assert_equal("\<c-v>52", getreginfo('x').regtype)
+
+  call setreg('y', ['two'], 'b52')
+  call setreg('y', #{regtype: 'v'})
+  call assert_equal("v", getreginfo('y').regtype)
+  call setreg('z', ['two'], 'b52')
+  call setreg('z', #{regtype: 'V'})
+  call assert_equal("V", getreginfo('z').regtype)
 
   bwipe!
 endfunc

```

**Comment by lacygoill** [comment]
I've applied the patch, used `setreg()` to reset the type of a register in all possible ways (characterwise to linewise, linewise to blockwise, ...), and everything seems to work fine.  I've even checked it worked as expected in one of my plugins (it did).  Thank you very much; I hope it will get merged.

**Comment by andymass** [comment]
@brammool, what do you think about the patch to set register type only?

**Comment by brammool** [comment]
I wonder how it can be useful to set the register type without setting the contents.  Normally they go together.

**Comment by andymass** [comment]
I sometimes have need to convert linewise registers into characterwise, e.g., when yanking a line then putting it on an existing line, or yanking a variable and putting it on a line of its own.  @lacygoill do you have a use-case for setreg with regtype only?

**Comment by lacygoill** [comment]
I have the same use-case.  I sometimes need to put a characterwise register on a new dedicated line.  [I do so by updating its type from characterwise to linewise](https://github.com/lacygoill/vim-brackets/blob/b17e7c97fcc178ee0be945aa840fa89d5802f587/autoload/brackets.vim#L480-L481):

    call getreginfo(reg_to_use)->extend({'regtype': 'l'})->setreg(reg_to_use)

It works, but it would be less verbose and more readable if `setreg()` accepted a dictionary whose only key is `regtype`:

    call setreg(reg_to_use, {'regtype': 'l'})

---

[Tpope does something similar](https://github.com/tpope/vim-unimpaired/blob/4afbe5ebf32ad85341b4c02b0e1d8ca96a64c561/plugin/unimpaired.vim#L336) in his `vim-unimpaired` plugin:

    call setreg(v:register, body, 'l')

**Comment by brammool** [comment]
OK, then it would be good to write the help for this, and cover the corner cases.  Should cover all possible types of contents and what happens when the register type is changed.  Esp. from/to block.

**Comment by lacygoill** [comment]
Just to be clear, the patch does not add any new feature.  We can already reset the type of a register from any type to any type.  For example, if the register `r` contains a characterwise text, and we want to reset its type to `^V123`, we can do so with:

    :call getreginfo('r')->extend({'regtype': 'b123'})->setreg('r')

The patch simply provides a simpler way to do the same thing:

    :call setreg('r', {'regtype': 'b123'})

That being said, I *think* I tested all the possibilities, and here is a summary of what I observed:

   1. if the register type is reset from characterwise to characterwise, or from linewise to linewise, no attribute of the register changes

   2. when the register is put from normal mode, if its type has been changed and reset to either characterwise or linewise, it's correctly put as a characterwise or linewise register

   3. when the register is put from command-line mode (via `:put`) or inserted via `CTRL-R`, the register is put as if it was linewise; which is equivalent to either `1.` or `2.`

   4. if the register type is reset to blockwise, and one of its line is shorter or longer than the specified block width, the line is neither truncated nor extended

   5. when the register is put from normal mode, if its type has been reset to blockwise, and one of its line is shorter than the specified block width, Vim adds as many trailing spaces as necessary to compensate if necessary

The tricky part to explain is the "if necessary".

Each line of the register is going to be put on some line of the buffer. And the cursor has a column position. Those 2 numbers *may* give a valid position in the buffer; one for each register line. I think Vim adds trailing spaces if, and only if, that position is valid and there exists a character in the buffer *after* it.

Since `1.` and `2.` don't describe anything unexpected, I'm not sure they need to be documented.
`3.` is already documented.
`4.` and `5.` may be unexpected, therefore need to be documented.
`4.` is easy to document, but `5.` is not.

Here is my first attempt:
```diff
diff --git a/runtime/doc/eval.txt b/runtime/doc/eval.txt
index ed4f8aab0..b17654b49 100644
--- a/runtime/doc/eval.txt
+++ b/runtime/doc/eval.txt
@@ -9265,8 +9265,20 @@ setreg({regname}, {value} [, {options}])
 		You can also change the type of a register by appending
 		nothing: >
 			:call setreg('a', '', 'al')
+<
+		Or by using a dictionary which includes the "regtype" key: >
 
-<		Can also be used as a |method|, the base is passed as the
+                        :call setreg('a', {'regtype': 'l'})
+<
+		Note: If you reset the type of a register to blockwise, and one
+		of its line is longer or shorter than the specified width of the
+		block, the line is neither truncated nor extended.  When you put
+		the register from normal mode, if one of its line is shorter
+		than the specified width of the block, Vim adds as many trailing
+		spaces as necessary to compensate, if necessary; that is, if
+		they are needed to reach an existing character in the buffer.
+
+		Can also be used as a |method|, the base is passed as the
 		second argument: >
 			GetText()->setreg('a')
 
```
@andymass: What is your opinion regarding these observations; did I miss anything?  And what about the help, can you rephrase it in a more understandable way?

**Comment by andymass** [comment]
@lacygoill Comparing to `setreg('a', 'some', ...)` is useful to get some exceptional cases.  I haven't tested these behaviors yet.

1. Setreg with linewise adds a trailing newline if there is none in the contents

```
:call setreg('a', "abc\n", 'l') | reg a
Type Name Content
  l  "a   abc^J
:call setreg('a', "abc", 'l') | reg a
Type Name Content
  l  "a   abc^J
:call setreg('a', ["abc"], 'l') | reg a
Type Name Content
  l  "a   abc^J
```
2 Blockwise removes one
```
:call setreg('a', "abc\n", 'b') | reg a
Type Name Content
  b  "a   abc
```

3. Block size doesn't really depend on contents

```
:call setreg('a', "abc\n", 'b123') | reg a | echo getregtype('a')
Type Name Content
  b  "a   abc
^V123
```

4. But if you don't specify block size, one will be picked automatically

```
:call setreg('a', "abc\n", 'b') | reg a | echo getregtype('a')
Type Name Content
  b  "a   abc
^V3

:call setreg('a', "abc\nxyzz", 'b') | reg a | echo getregtype('a')
Type Name Content
  b  "a   abc^Jxyzz
^V4
```

Given these behaviors, it seems as though `setreg('r', {'regtype': '...'})` _could_ or _should_ change the contents of the register.  In addition, if block is specified it needs to inspect the contents to derive a size.

If the current patch doesn't follow the above (I didn't test it yet), a simple alternative to do this would be to merely define

    :call setreg('r', {'regtype': TYPE})
       ==>
    :call getreginfo('r')->extend({'regtype': TYPE})->setreg('r')

at the cost of efficiency.

**Comment by lacygoill** [comment]
Oh I see; I misunderstood what was required.  I thought we only needed to document the usage of a dictionary.  But you're right; it would be better to document everything: dictionary, list, string.  I'll try to make more tests when I have the time, but it's not going to be easy, because there are so many parameters to consider:

   - which conversion type do we perform? (3 x 3 = 9 possibilities)
   - do we use a dictionary or a list/string as the second argument of `setreg()` (2 possibilities)
   - do we put from normal mode or insert/command-line mode (2 possibilities)

That's already 9 x 2 x 2 = 36 possibilities.
And then, there's the special case of resetting a register to blockwise:

   - are there some lines which are shorter than the specified size?  (2 possibilities)
   - are there some lines which are longer than the specified size?  (2 possibilities)
   - do we specify a size or not? (2 possibilities)

This means that each of the 12 previous possibilities which convert to blockwise:

   - characterwise to blockwise using a dictionary, but without a specified size
   - linewise to blockwise using a dictionary, but without a specified size
   - blockwise to blockwise using a dictionary, but without a specified size

   - characterwise to blockwise using a list/string, but without a specified size
   - linewise to blockwise using a list/string, but without a specified size
   - blockwise to blockwise using a list/string, but without a specified size

   - characterwise to blockwise using a dictionary, and with a specified size
   - linewise to blockwise using a dictionary, and with a specified size
   - blockwise to blockwise using a dictionary, and with a specified size

   - characterwise to blockwise using a list/string, and with a specified size
   - linewise to blockwise using a list/string, and with a specified size
   - blockwise to blockwise using a list/string, and with a specified size

must be multiplied by 2 x 2 x 2 = 8 additional possibilities.  Which makes 8 x 12 + 36 = 132 possibilities to test.

I did already make [some tests](https://gist.github.com/lacygoill/4f4922b19dea44497da9dc9ed962da7b), but nowhere near enough.

> Block size doesn't really depend on contents, and with a specified size

True, it doesn't change the register, but it changes what will be put in a buffer when using the `p` or `P` normal command.  I thought it should be documented, but I'm not sure anymore.
